internal/repository/sync.go

```diff
@@ -64,8 +67,33 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
+		return expectedRootKeys[i].KeyID < expectedRootKeys[j].KeyID
+	})
+
 	repository := &Repository{r: r}
 
+	slog.Debug("Verifying if root keys are expected root keys...")
+	if len(expectedRootKeys) > 0 {
+		state, err := policy.LoadFirstState(ctx, r)
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+		rootKeys, err := state.GetRootKeys()
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		if len(rootKeys) != len(expectedRootKeys) {
+			return repository, errors.New(ErrCloningRepository.Error() + ", " + "cloned root keys do not match the expected keys.")
+		}
+		for keyid := range rootKeys {
+			if !reflect.DeepEqual(rootKeys[keyid], expectedRootKeys[keyid]) {
+				return repository, errors.New(ErrCloningRepository.Error() + ", " + "cloned root keys do not match the expected keys.")
```

Comment:
Named error + errors.Join here and other instances?

Reply:
Fixed